#/usr/bin/env python
PACKAGE = "whud_vision"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Kp1",double_t,0,"PID's Kp for y",3,0,20)
gen.add("Ki1",double_t,0,"PID's Ki for y",0,0,20)
gen.add("Kd1",double_t,0,"PID's Kd for y",0.3,0,20)

gen.add("Kp2",double_t,0,"PID's Kp for x",3.0,0,20)
gen.add("Ki2",double_t,0,"PID's Ki for x",0,0,20)
gen.add("Kd2",double_t,0,"PID's Kd for x",0.3,0,20)

gen.add("H_low",int_t,0,"HSV's Low H",0,0,255)
gen.add("H_high",int_t,0,"HSV's High H",10,0,255)
gen.add("S_low",int_t,0,"HSV's Low H",43,0,255)
gen.add("S_high",int_t,0,"HSV's High H",255,0,255)
gen.add("V_low",int_t,0,"HSV's Low H",46,0,255)
gen.add("V_high",int_t,0,"HSV's High H",255,0,255)

gen.add("upper_area",int_t,0,"max area",20000,0,300000)
gen.add("lower_area",int_t,0,"mmin area",200,0,300000)

exit(gen.generate(PACKAGE,"whud_vision","whud_vision_dynamic"))